core/includes/update.inc

```diff
@@ -713,11 +713,11 @@ function update_finished($success, $results, $operations) {
 }
 
 /**
- * Returns a list of all the pending database updates.
+ * Returns a list of all the pending site updates.
  *
  * @return
- *   An associative array keyed by module name which contains all information
- *   about database updates that need to be run, and any updates that are not
+ *   An associative array keyed by module name, which contains all information
+ *   about site updates that need to be run and any updates that are not
```

Comment:
The punctuation in the existing comment is correct: The comma is not placed before *which*, since what follows is not a parenthetical; the comma before *run* and *and* is correct.

Reply:
Sorry, that's not correct. What follows is not a parenthetical, because it's not in parentheses.

One uses a comma before “which” when it introduces a nonrestrictive clause: a clause providing extra information that isn't essential to the sentence's meaning. If you drop the "which" clause, the sentence is still meaningful.

If you think that the clause is required, then it should be "that" (with no comma).

Reply:
A [parenthetical phrase](https://grammarist.com/grammar/parenthetical-phrases/) is introduced by comma, parentheses, or em dash.  
Using a comma before *which* is not required: [*In the Young Cook of Britain competition, the finalists were asked which famous person they would like to cook for.*](https://dictionary.cambridge.org/grammar/british-grammar/which)

Reply:
A comma before "which" is used when the clause is non-restrictive, which is the case in the PR (and also in this sentence).

https://www.scribbr.com/commas/comma-before-which/#:~:text=You%20need%20a%20comma%20before%20“which”%20when%20it%20introduces%20a,without%20changing%20the%20sentence%27s%20meaning.